src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexPersistence.cs

```diff
@@ -65,6 +66,9 @@ public class LuceneIndexPersistence : IDisposable
 
         internal LuceneVoronDirectory LuceneDirectory => _directory;
 
+        private readonly object _readersLock = new object();
```

Comment:
What is this used for? You are already using `DecRef` for concurrency, no?

Reply:
This used to sync the writes to `_lastReader`.
Since now we have 2 code paths that might modify it.